.circleci/config.yml

```diff
@@ -23,23 +23,45 @@ jobs:
       - run:
           name: Install dependencies
           command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r .circleci/requirements.txt
```

Comment:
I'm not against this change but what was the thought process? Also, did you have to set a root password to use sudo? Finally, I believe this installs directly into the docker container which (I believe) is re-used in future deployments. If we remove a dependency from requirements.txt, will the container still have the package installed? I could be thinking about this incorrectly... Let me know.

Reply:
Ah I was thinking we got a fresh container each time, but we're definitely hitting that cache folder so I guess that doesn't make sense. I'd like to test that on hello_kjmerf by installing some stuff with sudo and then seeing if it's still there on a separate run. If there's any doubt, let's use the virtual environment. All I was trying to do was save a few lines of config so nothing too noble there.

Reply:
I just did some some testing with [the CircleCI config](https://github.com/kjmerf/hello_kjmerf/blob/not-master/.circleci/config.yml) over at hello_kjmerf (not-master) branch, and here's what I believe now. For every build, a new  Docker container spins up on an existing VM. That docker container is fresh and clean. The install scripts run and if you leave a dependency out that you previously had in, even with caching on, then it will fail as the dependency will not be installed. I cannot find the cache with `ls -la` even though the path has been specified, so I'm guessing that the cache is stored on the host VM.

I think the sudo pip install is safe but if either one of you has hesitations I'm fine going back to the virtual environment.

Reply:
Fine by me. Good research.